tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -56,9 +58,8 @@ private[scalanative] object LLVM {
       // If pack2hashPrev is not empty, we don't recompile native library.
       // Even if native library changes(This is very rare case). If native library
       // changes, we should clean the project first.
-      if ((isLl || !Files.exists(objPath)) &&
-          (incCompilationContext.pack2hashPrev.isEmpty ||
-            incCompilationContext.isChanged(packageName))) {
+      if ((isLl || !Files.exists(objPath)) && incCompilationContext
+            .shouldCompile(packageName)) {
```

Comment:
Can we try to move this condition to separate line to make it more readable?
```suggestion
      if ((isLl || !Files.exists(objPath)) && 
          incCompilationContext.shouldCompile(packageName)) {
```